R/FilterStateLogical.R

```diff
@@ -188,11 +199,17 @@ LogicalFilterState <- R6::R6Class( # nolint
         return(NULL)
       }
       if (missing(dataname)) dataname <- private$dataname
+      varname <- private$get_varname_prefixed(dataname)
+      choices <- private$get_selected()
+      n_choices <- length(choices)
+
       filter_call <-
-        if (private$get_selected()) {
+        if (n_choices == 1 && choices) {
           private$get_varname_prefixed(dataname)
-        } else {
+        } else if (n_choices == 1 && !choices) {
           call("!", private$get_varname_prefixed(dataname))
+        } else {
+          call("%in%", varname, choices)
```

Comment:
shouldn't this be `call("%in%", private$get_varname_prefixed(dataname), choices)`?

Reply:
updated